### ISSUE TITLE
Add native XPC mechanism in addition to CLI

### ISSUE THREAD

**Issue by eugenesvk** [pqrs-org/Karabiner-Elements#3840]
Karabiner-Elements can accept variable changes from other apps via CLI commands, but then this isn't the most performant IPC mechanism to be used on potentially every key stroke.

Would it be possible for KE to also use some Mac-native mechanism for this (think it's this one https://developer.apple.com/documentation/xpc)

**Comment by eugenesvk** [comment]
.

**Comment by tekezo** [comment]
For that use case, you should use to.set_variable instead.
https://karabiner-elements.pqrs.org/docs/json/complex-modifications-manipulator-definition/to/set-variable/

**Comment by eugenesvk** [comment]
How does that work with an external app??? to.set_variable is an "internal" KE mechanism for setting vars

**Comment by tekezo** [comment]
Isn't the goal of this issue to change a variable on every keystroke?
If that's the case, then Karabiner-Elements built-in mechanisms should be sufficient.

**Comment by eugenesvk** [comment]
Yes, potentially, but the sender is still an **extrenal app**, so it's **not Karabiner** that's tracking the keystrokes, but another app that has a mechanism to run a command on every keystroke to convey information to Karabiner via changing a variable value, thus allowing Karabiner to change, e.g., "layers" of configuration depending on what's happening within that external app. The state in that app can potentially change on every keystroke.

For example, if you have a modal text editor, it can signal to KE that a specific mode is active by changing a variable `editor_this-mode_active=1` and KE would activate conditional remaps. And given that you can change the modes quickly (potentially on every keypress) you'd need a mechanism without all the lag involved with new CLI commands

**Comment by tekezo** [comment]
I see. I understand the request. Personally, there are times when I'd also like to change Karabiner-Elements behavior in response to state changes reported by a custom input source.
It seems like it should be possible to expose something through DistributedNotificationCenter.

**Comment by tekezo** [comment]
Considering both usability and security, adding `--set-variables-from-stdin` to karabiner_cli was the appropriate solution, so it has been added in the latest beta version (v16.1.16).

After launching the CLI as shown below, you can set variables by entering each JSON object on a single line. To communicate from other processes, use a pipe or a similar mechanism.

```shell
'/Library/Application Support/org.pqrs/Karabiner-Elements/bin/karabiner_cli' --set-variables-from-stdin
```

```shell
{"hello":1,"world":"a"}
{"hello":2,"world":"b"}
{"hello":3,"world":"c"}
```
